Sequilize integration

dmrado opened the pull request:
отредактировал модели для добавления foreignKey
попытался создать seeders

abi83 committed:
some work

abi83 committed:
config

abi83 committed:
minimum working version

abi83 committed:
db sync

Commit:
import { config } from 'dotenv';   "type": "module",

Commit:
next.config.js заменен экспорт в соответствии с "type": "module",

Commit:
next.config.js не дает собираться приложению, а tailwind.config.js не работает с "type": "module". А без этойдирективы собирается и работает

Commit:
seeder post seeder

abi83 commented on .gitignore:
```diff
@@ -34,3 +34,4 @@ yarn-error.log*
 # typescript
 *.tsbuildinfo
 next-env.d.ts
+/src/app/db/.env
```
.env файл лучше хранить в корне проекта (он дожен быть один).
Он используется ведь не только для базы данных?

Что касается гит-игнор, то лучше прописывать енв файл по маске типа такого: https://stackoverflow.com/questions/59055847/ignoring-env-files-in-my-project-with-gitignore

abi83 commented on next.config.js:
У меня в такой конфигурации не работает. Что насчет этого? https://nextjs.org/docs/pages/api-reference/next-config-js

abi83 commented on package.json:
```diff
@@ -2,27 +2,31 @@
   "devDependencies": {
     "autoprefixer": "^10.0.1",
     "postcss": "^8",
+    "sequelize-cli": "^6.6.2",
```
А нам эта зависимость вообще нужна?

abi83 commented on src/app/db/connection.js:
```diff
@@ -1,11 +1,21 @@
+import mysql2 from 'mysql2'
 
-export default new Sequelize({
+// todo: load secrets from env file
```
С енв файлом не справился?

abi83 commented on src/app/db/post.model.js:
```diff
@@ -1,20 +1,24 @@
+      type: DataTypes.STRING
+    },
+    userId: {
+    type: DataTypes.STRING,
```
id обычно INTEGER

abi83 commented on src/app/db/post.model.js:
```diff
@@ -1,20 +1,24 @@
+    type: DataTypes.STRING,
+        references: {
+            model: User,
+            key: 'userId'
```
лучше все таки отсылаться к полю id, которое создается автоматически.

abi83 commented on src/app/db/post.seeder.js:
```diff
@@ -0,0 +1,31 @@
+                title: 'Title Post-2',
+                text: 'Text Post-2, Text Post-2, Text Post-2,Text Post-2, Text Post-2,Text Post-2',
+                // todo требует default value, не знаю как по-другому передать его сюда
+                createdAt: new Date(),
```
https://sequelize.org/docs/v6/other-topics/migrations/#creating-the-first-seed

Ну похоже это правильно.... Хотя я бы ожидал что это может быть сделано на уровне базы.... Но видимо mysql в это не может: https://stackoverflow.com/questions/168736/how-do-you-set-a-default-value-for-a-mysql-datetime-column/10603198#10603198

abi83 committed:
some preparation

Commit:
MWP without css

Commit:
Update page.js

попытка поймать post.id в серверном компоненте без использования обработчика события

Commit:
foo()

так тоже не работает

Commit:
css отключен из-за конфликта импортов

Commit:
установка typeScript и esLint

Commit:
попытка создания types

abi83 committed:
next14ts new project created

abi83 committed:
not successfull typescript sequelize. mysql2 is not working

abi83 committed:
old version removed

abi83 committed:
minor fixes

Commit:
connection с данными Дмитрия

Commit:
PostPAge.tsx

abi83 committed:
css attached

Commit:
type for children on layout & .env .gitignore

Commit:
secrets

Commit:
seeders need fix

Commit:
не понятно как верстать на гридах в tailwind

Commit:
PostsPage верстка

в post.model.ts добавлено свойство updatedAt для отображения на боди карточки поста

Commit:
Страница одного поста готова

Commit:
posts/new верстка

  revalidatePath('/posts') не работает

Commit:
post [id]

Commit:
попытка сделать жесткий DeletePost

Commit:
404 и установка пакетов которых якобы не хватает на post[id]

Commit:
DeletePost is ready!

abi83 committed:
some work during the call
